### PR TITLE
 Add basic authentication capabilities to Pulsar SQL

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Allow the presto connector to authentication with a Pulsar Cluster.

### CODE REVIEW

**Comment by jerrypeng** [comment]
rerun cpp tests

**Comment by jerrypeng** [comment]
rerun integration tests

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarSplitManager.java [hunk 1]
```diff
@@ -143,6 +148,10 @@ ManagedLedgerFactory getManagedLedgerFactory() throws Exception {
         try {
             numPartitions = (this.pulsarAdmin.topics().getPartitionedTopicMetadata(topicName.toString())).partitions;
         } catch (PulsarAdminException e) {
+            if (e.getStatusCode() == 401) {
+                throw new PrestoException(QUERY_REJECTED, "Unauthorized");
```
Comment (sijie) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarSplitManager.java]: Can you improve the error message? For example, adding the topic name to the error message.

**Comment by jerrypeng** [comment]
@sijie PTAL

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarMetadata.java [hunk 2]
```diff
@@ -291,7 +301,12 @@ private ConnectorTableMetadata getTableMetadata(SchemaTableName schemaTableName,
             if (e.getStatusCode() == 404) {
                 // to indicate that we can't read from topic because there is no schema
                 return null;
+            } else if (e.getStatusCode() == 401) {
+                throw new PrestoException(QUERY_REJECTED,
+                        String.format("Failed to get schema information for topic %s/%s: Unauthorized",
```
Comment (sijie) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarMetadata.java]: nit: schema has a different meaning in pulsar. so I would suggest being specific here: "Failed to get pulsar schema information"

File: pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarSplitManager.java [hunk 3]
```diff
@@ -105,6 +106,12 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHand
             schemaInfo = this.pulsarAdmin.schemas().getSchemaInfo(
                     String.format("%s/%s", namespace, tableHandle.getTableName()));
         } catch (PulsarAdminException e) {
+            if (e.getStatusCode() == 401) {
+                throw new PrestoException(QUERY_REJECTED,
+                        String.format("Failed to get schema for topic %s/%s: Unauthorized",
```
Comment (sijie) [inline pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/PulsarSplitManager.java]:
```suggestion
                        String.format("Failed to get pulsar schema for topic %s/%s: Unauthorized",
```

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by sijie** [comment]
run cpp tests

**Comment by sijie** [comment]
run cpp tests

**Comment by sijie** [comment]
run cpp tests

**Comment by jiazhai** [comment]
mark this as 2.4.1 for conflict #4890

**Pull request merged**